### DISCUSSION TITLE
Clearer documentation/errors for enabling client routing

### DISCUSSION THREAD

**Discussion by dylmye** [vikejs/vike#1133] [Discussion]
Hi there! I was sent from the community Discord. Thanks for all of your hard work on this project - it's fantastic.

I tried enabling client routing and found the documentation to be unclear.

I started from the ["server routing vs client routing" guide document](https://vite-plugin-ssr.com/server-routing-vs-client-routing#client-routing). I then clicked [the docs link](https://vite-plugin-ssr.com/clientRouting) with the words "setting the `clientRouting` option option to `true`." (also note the duplicate wording "option" :)

I followed step 1 which was easy, enabling the value on client side routing.

For step 2, it [links to](https://vite-plugin-ssr.com/render-client) the "render() hook (client-side)" page. However this page makes no mention of `isHydration` or `createRoot()`. Looking at the example on the clientRouting page shows how to do it but I think it could be better exposed by amending the step 2 text:

> Update and adapt the [client-side render() hook](https://vite-plugin-ssr.com/render-client) **to conditionally hydrate the root**

or something like that. Additionally, there is a [better explained doc](https://vite-plugin-ssr.com/render-modes) with the client-side routing render code that could possibly be better to link to.

Please let me know what you lot think!

**Comment by lourot** [comment]
Hey @dylmye :wave: Thanks for the feedback. It all makes sense and I'll create a doc PR to improve this area today.

**Reply by brillout** [reply]
@dylmye We just talked about it with @AurelienLourot.

We agree, the docs could be improved. The plan is to have [vike-react](https://github.com/vikejs/vike-react) provide a much easier getting started (it includes Client Routing and an `ssr: boolean` flag which seems to be the two things you're looking for). The docs will then actually be simplified and reduced as it will asume you're using a `vike-*` package. (We still want to keep and improve the existing documentation but it will be moved to a section dedicated to users who want to manually integrate the ui framework themselves.)

Vike React is in beta, but it's fairly stable already. I recommend using it while letting us know if you stumble upon any issues.

> duplicate wording

PR welcome :)

**Reply by dylmye** [reply]
Thank you both 🙏

**Comment by lourot** [comment]
I have created #1135

**Reply by dylmye** [reply]
It looks good, appreciate your effort and assistance on this :)
